feat(windows): allow setting window class name, closes #769

geraudloup opened the pull request:
allow to customize the window class name instead of current value hard coded "Window Class"

Currently, on Windows, all windows created by Tao have the same class Window Class, it would be neet to allow for customization of the Windows' window class name. One use case: some windows custom hotkey managers leverage window class names to select windows...

Waiting for merge in upstream project winit (pass all 55 checks) : https://github.com/rust-windowing/winit/pull/2978
(closing mirrored issue https://github.com/rust-windowing/winit/issues/2977)

<!--
Update "[ ]" to "[x]" to check a box

Please make sure to read the Pull Request Guidelines: https://github.com/tauri-apps/tauri/blob/dev/.github/CONTRIBUTING.md#pull-request-guidelines
-->

### What kind of change does this PR introduce?
<!-- Check at least one. If you are introducing a new binding, you must reference an issue where this binding has been proposed, discussed and approved by the maintainers. -->

- [ ] Bugfix
- [X] Feature
- [ ] Docs
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?
<!-- If yes, please describe the impact and migration path for existing applications in an attached issue. -->

- [ ] Yes
- [X] No

### Checklist
- [X] This PR will resolve #769 
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tao/blob/dev/.changes/readme.md).
- [X] I have added a convincing reason for adding this feature, if necessary
- [X] It can be built on all targets and pass CI/CD.

geraudloup committed:
On Windows, set name of Window Class, closes #769
allow to customize it instead of current value hard coded "Window Class"

amrbashir commented on src/platform_impl/windows/window.rs:
```diff
@@ -1180,8 +1180,8 @@ unsafe fn init<T: 'static>(
 
-unsafe fn register_window_class() -> Vec<u16> {
-  let class_name = util::encode_wide("Window Class");
+unsafe fn register_window_class<T: 'static>(window_classname: &String) -> Vec<u16> {
```
```suggestion
unsafe fn register_window_class(window_classname: &str) -> Vec<u16> {
```

amrbashir commented on src/platform_impl/windows/window.rs:
```diff
@@ -996,7 +996,7 @@ unsafe fn init<T: 'static>(
 ) -> Result<Window, RootOsError> {
   // registering the window class
-  let class_name = register_window_class();
+  let class_name = register_window_class::<T>(&pl_attribs.window_classname);
```
```suggestion
  let class_name = register_window_class(&pl_attribs.window_classname);
```

amrbashir reviewed (changes requested):
Thanks for your contribution, could you also add a changefile in .changes directory?

geraudloup committed:
Update src/platform_impl/windows/window.rs

Co-authored-by: Amr Bashir <amr.bashir2015@gmail.com>

geraudloup committed:
Update src/platform_impl/windows/window.rs

Co-authored-by: Amr Bashir <amr.bashir2015@gmail.com>

geraudloup committed:
On Windows, set name of Window Class closes #769
allow to customize it instead of current value hard coded "Window Class"

geraudloup commented:
> Thanks for your contribution, could you also add a changefile in .changes directory?

Yes ! I added one (following the exemple of `windows-ltr.md`) and tested and accepted your suggested changes (I also updated my PR on winit with your improvements) !

amrbashir commented:
@wusyong should this API be ported to the v0.16 branch or wait until tauri v2?

amrbashir commented:
cherry picked into v0.16 in https://github.com/tauri-apps/tao/commit/ae664f05daf10456b4721214372bb5409abe53b9
